src/x11/x.c

```diff
@@ -584,14 +584,56 @@ static dimension_t x_render_layout(cairo_t *c, colored_layout *cl, colored_layou
         int bg_height = MAX(settings.notification_height, (2 * settings.padding) + h);
         double bg_half_height = settings.notification_height/2.0;
         int pango_offset = (int) floor(h/2.0);
+        const float degrees = M_PI / 180.0;
 
         if (first) bg_height += settings.frame_width;
         if (last) bg_height += settings.frame_width;
         else bg_height += settings.separator_height;
 
-        cairo_set_source_rgb(c, cl->frame.r, cl->frame.g, cl->frame.b);
-        cairo_rectangle(c, bg_x, bg_y, bg_width, bg_height);
-        cairo_fill(c);
+        XWindowAttributes win_attr;
+        XGetWindowAttributes(xctx.dpy, xctx.win, &win_attr);
```

Comment:
I'd like to see, that we extract the width from cairo instead of X. Regarding wayland support, this will be much easier to port.